packages/workers-playground/src/QuickEditor/HTTPTab/HTTPTab.tsx

```diff
@@ -89,22 +93,33 @@ export function HTTPTab() {
 					setIsLoading(false);
 				}
 			}
-		},
-		[previewHash, setPreviewUrl, previewUrl, method, headers, hasBody, body]
-	);
-	const ensureDevtoolsConnected = useRef(false);
-	useEffect(() => {
-		if (!ensureDevtoolsConnected.current && previewHash && !isLoading) {
-			void onSendRequest();
-			ensureDevtoolsConnected.current = true;
 		}
-	}, [previewHash, isLoading, onSendRequest]);
+
+		void sendRequest();
+	}, [
+		refreshTimestamp,
+		previewHash,
+		previewUrl,
+		method,
+		headers,
+		hasBody,
+		body,
+	]);
 
 	return (
 		<Div display="flex" flexDirection="column" width="100%">
 			<Form
 				display="flex"
-				onSubmit={(e) => void onSendRequest(e)}
+				onSubmit={(e) => {
+					e.preventDefault();
+					preview();
+
+					if (url === previewUrl) {
+						setRefreshTimestamp(new Date().toISOString());
```

Comment:
The timestamp has no use but trigger the effect above (`sendRequest`) to re-run.